core/src/main/java/com/sequenceiq/cloudbreak/service/image/PreWarmParcelParser.java

```diff
@@ -25,31 +25,48 @@ public class PreWarmParcelParser {
     private CsdParcelNameMatcher csdParcelNameMatcher;
 
     public Optional<ClouderaManagerProduct> parseProductFromParcel(List<String> parcel, List<String> csdList) {
-        Optional<String> url = parcel.stream().filter(parcelPart -> parcelPart.startsWith("http://") || parcelPart.startsWith("https://"))
-                .findFirst();
-        Optional<String> nameAndVersion = parcel.stream().filter(parcelPart -> parcelPart.endsWith(".parcel"))
-                .findFirst();
-        if (url.isEmpty() || nameAndVersion.isEmpty()) {
+        Optional<String> baseUrl = getBaseUrl(parcel);
+        Optional<String> nameAndVersion = getNameAndVersion(parcel);
+        if (baseUrl.isEmpty() || nameAndVersion.isEmpty()) {
             LOGGER.warn("Parcel URL or (name and version) could not be found in the image metadata. "
-                    + "Parcel url: '{}', Parcel name and version: '{}'.", url.orElse("null"), nameAndVersion.orElse("null"));
+                    + "Parcel baseUrl: '{}', Parcel name and version: '{}'.", baseUrl.orElse("null"), nameAndVersion.orElse("null"));
             return Optional.empty();
         } else {
             ClouderaManagerProduct product = new ClouderaManagerProduct();
             String name = substringBefore(nameAndVersion.get(), "-");
             product.setName(name);
             product.setDisplayName(nameAndVersion.get());
-            String version = substringAfter(substringBeforeLast(nameAndVersion.get(), "-"), "-");
-            product.setVersion(version);
-            LOGGER.debug("The parsed product name for parcel: '{}', version: '{}'. URL: '{}'",  name, version, url.get());
-            product.setParcel(url.get());
+            product.setVersion(substringAfter(substringBeforeLast(nameAndVersion.get(), "-"), "-"));
+            product.setParcel(baseUrl.get());
+            product.setParcelFileUrl(removeUnnecessaryCharacters(baseUrl.get()).concat("/").concat(nameAndVersion.get()));
             product.setCsd(collectCsdParcels(csdList, name));
+            LOGGER.debug("The parsed product {}", product);
             return Optional.of(product);
         }
     }
 
+    private Optional<String> getNameAndVersion(List<String> parcel) {
+        return parcel.stream()
+                .filter(parcelPart -> parcelPart.endsWith(".parcel"))
+                .findFirst();
+    }
+
+    private Optional<String> getBaseUrl(List<String> parcel) {
+        return parcel.stream()
+                .filter(parcelPart -> parcelPart.startsWith("http://") || parcelPart.startsWith("https://"))
+                .findFirst();
+    }
+
     private List<String> collectCsdParcels(List<String> csdList, String name) {
         return csdList.stream()
                 .filter(csd -> csdParcelNameMatcher.matching(csd, name))
                 .collect(Collectors.toList());
     }
+
+    private String removeUnnecessaryCharacters(String baseUrl) {
```

Comment:
wouldn't `StringUtils.stripEnd(basurl, "./")` be sufficient here?
I'm not really seeing any related unit test for this, so I'm not sure why was this introduced